### PR TITLE
Helpful assertion for isAlwaysShown error

### LINKED ISSUE #53771: Scrollbar always shown property bugs when set to true
When setting Scrollbar isAlwaysShown to true the following bugs are observed.
1. The scrollbar will only show and persist after scrolling at least once using the mouse wheel or by dragging.

2. If the application is in debug mode and the isAlwaysShown property is changed by the developer from false to true, a hot reload will cause an assertion error to be thrown.

Both feature and bug introduced in [PR50752](https://github.com/flutter/flutter/pull/50752)
cc\ @justinmc @onatcipli 

## Steps to Reproduce
Reproduced on Desktop Web
Reproduced on Desktop Windows
```
@override
  Widget build(BuildContext context) {
    return Material(
      color: Colors.white,
      child: Scrollbar(
        isAlwaysShown: true, // Toggle then hot reload
        child: ListView(
          children: <Widget>[
            for (int i = 0; i < 20; i++)
              ListTile(
                leading: CircleAvatar(
                  child: FlutterLogo(size: 20),
                ),
                title: Text("Jon Do"),
                subtitle: Text("Some Text"),
                trailing: Text("Other Text"),
                onTap: () {},
              )
          ],
        ),
      ),
    );
  }
```

<details>
  <summary>Logs</summary>

```
[√] Flutter (Channel master, v1.16.4-pre.33, on Microsoft Windows [Version 10.0.18363.720], locale en-CA)
    • Flutter version 1.16.4-pre.33 at D:\flutter
    • Framework revision aee9e94c21 (2 days ago), 2020-03-30 16:12:40 -0700
    • Engine revision c779894d08
    • Dart version 2.8.0 (build 2.8.0-dev.18.0 1402e8e1a4)

 
[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\kusay\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_SDK_ROOT = C:\Users\kusay\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Professional 2019 16.3.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional
    • Visual Studio Professional 2019 version 16.3.29424.173

[!] Android Studio (version 3.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] VS Code, 64-bit edition (version 1.43.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.8.1

[√] Connected device (3 available)
    • Windows    • Windows    • windows-x64    • Microsoft Windows [Version 10.0.18363.720]
    • Chrome     • chrome     • web-javascript • Google Chrome 80.0.3987.149
    • Web Server • web-server • web-javascript • Flutter Tools

! Doctor found issues in 1 category.
```

```
```

```
```

</details>

### PR DESCRIPTION
**Pull request by justinmc**
## Description

When a scrollbar's isAlwaysShown is true, it must have a controller that's attached to a scrollview. Previously, an assertion deeper in the framework would get triggered if this wasn't the case, with an error message that's not so informative.  This PR adds an assertion directly in the constructor with a more helpful error message.

## Related Issues

Closes https://github.com/flutter/flutter/issues/53771

## Tests

I added the following tests:
  * Check that an assertion is triggered when no controller at all.
  * Check that an assertion is triggered when the controller isn't attached to a scroll view.
  * The two above tests for material and cupertino.

### CODE REVIEW

**Comment by justinmc** [comment]
@onatcipli Does this look good to you?  I wanted to make sure people understand they need to pass a controller to avoid confusion like in https://github.com/flutter/flutter/issues/53771#issuecomment-635723884.

File: packages/flutter/test/material/scrollbar_test.dart [hunk 1]
```diff
@@ -201,6 +201,56 @@ void main() {
     expect(scrollbar.controller, isNotNull);
   }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS,  TargetPlatform.macOS }));
 
+  testWidgets('When isAlwaysShown is truew, must pass a controller',
```
Comment (LongCatIsLooong) [inline packages/flutter/test/material/scrollbar_test.dart:204]:
```suggestion
  testWidgets('When isAlwaysShown is true, must pass a controller',
```

File: packages/flutter/test/cupertino/scrollbar_test.dart [hunk 2]
```diff
@@ -164,6 +164,58 @@ void main() {
     await tester.pump(_kScrollbarFadeDuration);
   });
 
+  testWidgets('When isAlwaysShown is truew, must pass a controller',
```
Comment (LongCatIsLooong) [inline packages/flutter/test/cupertino/scrollbar_test.dart:167]:
```suggestion
  testWidgets('When isAlwaysShown is true, must pass a controller',
```

File: packages/flutter/test/cupertino/scrollbar_test.dart [hunk 3]
```diff
@@ -164,6 +164,58 @@ void main() {
...
+                height: 4000.0,
+              ),
+            ),
+          ),
+        ),
+      );
+    }
+
+    expect(() async {
+      await tester.pumpWidget(viewWithScroll());
```
Comment (LongCatIsLooong) [inline packages/flutter/test/cupertino/scrollbar_test.dart:188]: Will this test pass prior to the change? Before this change the assert should still trigger at the end of the frame right?

**Review by LongCatIsLooong — APPROVED** [review]
LGTM!

File: packages/flutter/test/cupertino/scrollbar_test.dart (hunk 3 above)
Comment (justinmc) [reply packages/flutter/test/cupertino/scrollbar_test.dart:188]: I just tried it and it fails prior to the change (so does the corresponding material test).

It does still throw an error though, but I guess it's happening in the post frame callback and so it's not caught by this.

**Pull request merged**